PelicanPlatform/pelican
# Replace ComputeExternalAddress by param Server.ExternalWebUrl and rename Server.Port and Server.Address

Closes #372 

The `ComputeExternalAddress()` seems to be less useful if we simply assigned `Server.Hostname`+`Server.Port` as the default return value of the function, so I replace the function by directly setting the default value of `Server.ExternalAddress`, and, of course, maintaining the url type.

Later while I making the changes to the codebase, I figured that those three config names (`Server.Port`, `Server.Address`, `Server.ExternalAddress`) are very broad and can cause confusion when people try to use them. So I decided to rename them to better reflect what they meant:

* `Server.Port` -> `Server.WebPort`: In our codebase, `Server.Port` is used to setup web engine and serves as the port for various web related services (Prometheus, IssuerURL, etc). So I figured it's better to call it `Server.WebPort` to be dinstinct with `XrootD.Port`
* `Server.Address` -> `Server.WebHost`: `Server.Address` is only used when we setup web engine so it's better call it `WebHost`
* `Server.ExternalAddress` -> `Server.ExternalWebUrl`: per description, this value is an URL that can represent the server's web service for external users, so it makes sense to name it as what it's intended to do.

**This is a potential breaking change, but given the fact that we haven't started distributing our server instances, it shouldn't hurt too much if we put it in the release note.**

## Testing Instructions

Run your servers and go to `origin` and the web ui can load as expected. Note your console output for all the servers and there shouldn't be any errors. Now change the `Server.ExternalAddress` to `https://${Server.Hostname}:${Server.Port}` where the ${} is the value of the config names, and re-run all servers and they should still work as expected.
